module-system/module-implementations/sov-blob-storage/src/capabilities.rs

```diff
@@ -0,0 +1,71 @@
+use sov_modules_api::capabilities::{BlobRefOrOwned, BlobSelector};
+use sov_modules_api::{Context, Spec};
+use sov_rollup_interface::da::BlobReaderTrait;
+use sov_state::WorkingSet;
+
+use crate::BlobStorage;
+
+impl<C: Context> BlobSelector for BlobStorage<C> {
+    type Context = C;
+
+    fn get_blobs_for_this_slot<'a, I, B>(
+        &self,
+        current_blobs: I,
+        working_set: &mut WorkingSet<<Self::Context as Spec>::Storage>,
+    ) -> anyhow::Result<Vec<BlobRefOrOwned<'a, B>>>
+    where
+        B: BlobReaderTrait,
+        I: IntoIterator<Item = &'a mut B>,
+    {
+        // TODO: Chain-state module: https://github.com/Sovereign-Labs/sovereign-sdk/pull/598/
+        let current_slot: u64 = self.get_current_slot_number(working_set);
+        let past_deferred: Vec<B> = current_slot
+            .checked_sub(self.get_deferred_slots_count(working_set))
+            .map(|pull_from_slot| self.take_blobs_for_block_number(pull_from_slot, working_set))
+            .unwrap_or_default();
+        let preferred_sequencer = self.get_preferred_sequencer(working_set);
+
+        if preferred_sequencer.is_none() {
+            return Ok(current_blobs
+                .into_iter()
+                .map(Into::into)
+                .chain(past_deferred.into_iter().map(Into::into))
+                .collect());
+        }
+
+        let preferred_sequencer = preferred_sequencer.unwrap();
```

Comment:
this and line 28 can be replaced with `if let  preferred_sequencer = Some(..) else {..}`